### PR TITLE
[FEATURE] - Add unit tests

### PR DESCRIPTION
**Pull request by yossi-eynav**
cc: 
@ealush

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ class Uploader {
 
         const lastDot = name.lastIndexOf('.'),
             nameLength = name.length,
-            hasExtension = (nameLength - lastDot) <= 5;
+            hasExtension = lastDot > 0 && (nameLength - lastDot) <= 5;
```
Comment (ealush) [inline src/index.js:38]: Good catch. Didn't think of hidden files

File: src/index.js [hunk 2]
```diff
@@ -51,6 +51,8 @@ class Uploader {
     }
 
     static prettyFileSize(size) {
+        if(!size) { return '0KB'; }
```
Comment (ealush) [inline src/index.js:54]: This seems more like an error to me. Maybe we should throw here?

File: src/index_spec.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+describe('Uploader', () => {
+    let uploader, file, fileList;
+
+    beforeEach(() => {
+        file = new FileAPI.File({ 
+            name: "abc-song.txt",
+            type: "text/plain",
+            buffer: new Buffer("abcdefg,hijklmnop, qrs, tuv, double-u, x, y and z")
+          });
```
Comment (ealush) [inline src/index_spec.js]: Style: indentation

File: src/index_spec.js [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+const {describe} = require('mocha');
+
+global.FormData = require('form-data');
+global.XMLHttpRequest = sinon.useFakeXMLHttpRequest();
+
+describe('Uploader', () => {
+    let uploader, file, fileList;
+
+    beforeEach(() => {
+        file = new FileAPI.File({
```
Comment (ealush) [inline src/index_spec.js]: I would add multiple files with different settings as an array, and run all tests over each element in the array.

File: src/index_spec.js [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        });
+
+        it('should ignore special characters', () => {
+            expect(Uploader.itemId('!!1@#$@#!$.jpg').startsWith('file_1jpg')).to.equal(true);            
+        });
+    });
+
+    describe('Pretty File Size', () => {
+        it('Default value should be 0KB', () => {
+            expect(Uploader.prettyFileSize()).to.equal('0KB');
```
Comment (ealush) [inline src/index_spec.js:45]: expect(...).to.throw(...)

Imo. If the default is just for emulated files, it should be made an error instead.

File: src/index_spec.js [hunk 6]
```diff
@@ -0,0 +1,134 @@
+const sinon = require('sinon');
+const Uploader = require('./index.js');
+const FileAPI = require('file-api')
+const {expect} = require('chai');
+const {describe} = require('mocha');
+
+global.FormData = require('form-data');
```
Comment (ealush) [inline src/index_spec.js:7]: Some of the describes can go in another files. Some of them don't really use the new file you create and don't need the same beforeEach

File: src/index.js (hunk 2 above)
Comment (yossi-eynav) [reply src/index.js:54]: lets talk about it when you come back

File: src/index_spec.js (hunk 6 above)
Comment (yossi-eynav) [reply src/index_spec.js:7]: You're right, fixed it 👍

File: src/index_spec.js (hunk 4 above)
Comment (yossi-eynav) [reply src/index_spec.js]: why?

File: src/index_spec.js (hunk 5 above)
Comment (yossi-eynav) [reply src/index_spec.js:45]: lets talk about it offline

File: src/index_spec.js (hunk 4 above)
Comment (ealush) [reply src/index_spec.js]: So you can test different cases.

**Pull request closed without merging**